plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1420,7 +1446,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
       }
 
       chain.start_block( block_time, blocks_to_confirm, features_to_activate );
-   } FC_LOG_AND_DROP();
+   } LOG_AND_DROP();
```

Comment:
This will lead to `producer_plugin_impl::start_block()` rethrowing on a `std::bad_alloc` which in turn leads to `producer_plugin_impl::schedule_production_loop()` throwing and I'm pretty sure thats unexpected behavior in the rest of the code.  

We should probably figure out what to do with `std::bad_alloc` and change `LOG_AND_DROP` to properly do that in all of these cases.

Reply:
Why is this `LOG_AND_DROP` here? Seems like any exception from `start_block` should not be ignored.

Reply:
outside of the "special cases" that result in app-terminating behavior like bad_allocs the other exceptions would result in `chain.is_building_block()` on the next line returning `false` ultimately leading to a return of `start_block_result::failed` 

I agree this could be handled better but the callers are not expecting exceptions thrown out of this (we may even want to throw on `noexcept` to make that obvious) so, they all need to be "handled" in some way and reflected in the returned value. 

This is important for any "recoverable" error as this is usually called inside some async task handler, as part of a destructor for an `fc::scoped_exit` that calls `schedule_production_loop()` in its destructor.  Even if we handled it better, throwing exceptions into the `io_service` is not going to be the right move so, we'd be swallowing it somewhere.

Reply:
`scoped_exit()` does make it "interesting". Ok, I added he `is_quiting()` check which will return `failed` now.